mapper-runtime/src/main/java/com/datastax/oss/driver/internal/mapper/entity/EntityHelperBase.java

```diff
@@ -75,4 +86,70 @@ protected void throwIfKeyspaceMissing() {
               DaoKeyspace.class.getSimpleName()));
     }
   }
+
+  public List<CqlIdentifier> findMissingColumns(
+      List<CqlIdentifier> entityColumns, Collection<ColumnMetadata> cqlColumns) {
+    return findMissingColumnsCql(
+        entityColumns,
+        cqlColumns.stream().map(ColumnMetadata::getName).collect(Collectors.toList()));
+  }
+
+  public List<CqlIdentifier> findMissingColumnsCql(
+      List<CqlIdentifier> entityColumns, Collection<CqlIdentifier> cqlColumns) {
+    List<CqlIdentifier> missingColumns = new ArrayList<>();
+    for (CqlIdentifier entityCqlIdentifier : entityColumns) {
+      if (!cqlColumns.contains(entityCqlIdentifier)) {
+        missingColumns.add(entityCqlIdentifier);
+      }
+    }
+    return missingColumns;
+  }
+
+  /**
+   * When the new instance of a class annotated with {@link Dao} is created an automatic check for
+   * schema validation is performed. It verifies if all {@link Dao} entity fields are present in CQL
+   * table. If not the {@link IllegalArgumentException} exception with detailed message is thrown.
+   * This check has startup overhead so once your app is stable you may want to disable it. The
+   * schema validation check is enabled by default. It can be disabled using the {@link
+   * MapperBuilder#withSchemaValidationEnabled(boolean)} method.
+   */
+  public abstract void validateEntityFields();
+
+  public static List<String> findMissingTypes(
```

Comment:
Nit: it's not really that the types are "missing", more that they don't match. Maybe `findTypeMismatches`?